utils/src/resource_location.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "utils/resource_location.h"
+
+#include <filesystem>
+#include <regex>
+
+#include "network/uri/uri.hpp"
+
+#include "utils/path.h"
+#include "utils/path_conversions.h"
+#include "utils/platform.h"
+
+namespace hlasm_plugin::utils::resource {
+namespace {
+struct dissected_uri
+{
+    std::optional<std::string> scheme = std::nullopt;
+    std::optional<std::string> host = std::nullopt;
+    std::optional<std::string> auth = std::nullopt;
+    std::optional<std::string> path = std::nullopt;
+    std::optional<std::string> query = std::nullopt;
+    std::optional<std::string> fragment = std::nullopt;
+};
+
+std::string format_path(std::string hostname, std::string path)
+{
+    if (!hostname.empty())
```

Comment:
I am not 100% sure, but I think this code would be much clearer if it was split into the windows/non-windows part.

Reply:
Refactored a bit and moved to a different file (it didn't feel right to have it in `resource_location`)